subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -670,6 +670,51 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
         file("build/classes/java/main/io/example/Example.class").exists()
     }
 
+    @Requires(adhoc = {AvailableJavaHomes.getJdk(VERSION_1_9)})
```

Comment:
Maybe also add `@Issue("https://github.com/gradle/gradle/issues/2537")`

For the Requires annotation: I wonder if it would be better to annotate it with `TestPrecondition.JDK9_OR_LATER`. Now the test will run in all different environments on CI (Java7, Java8, Java9, ...) but will always use the Java9 version it picks itself. So effectively it will be the same environment every time. Which kind of also defeats the point of defining the Java9 version via the `JAVA_9` env variable.